modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java

```diff
@@ -285,6 +301,20 @@ public final URL findResource(final String name) {
           }
         }
       }
+    } else if (name.endsWith(CLASS_EXTENSION)) {
```

Comment:
I think this also affects `getResources`

Reply:
It does, getResources delegates to findResources. There's a test cases validating that scenario.